### PR TITLE
Add auth middleware and tokens to delivery employee requests

### PR DESCRIPTION
**Pull request by LMBishop**
image

### CODE REVIEW

**Comment by LMBishop** [comment]
image

File: app.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ import path from 'path';
 import { deliveryController } from './controller/deliveryController';
 import { authController } from './controller/authController';
 import { salesemployeeController } from './controller/salesemployeeController';
+import authMiddleware from './middleware/auth';
+import { ActiveSession } from './model/auth';
```
Comment (TomBillingtonUK) [inline app.ts:9]: import type

File: middleware/auth.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { NextFunction, Request, Response } from "express";
```
Comment (TomBillingtonUK) [inline middleware/auth.ts:1]: import type

File: service/authService.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
-import { Credentials, User } from "../model/auth";
+import { ActiveSession, Credentials, User } from "../model/auth";
```
Comment (TomBillingtonUK) [inline service/authService.ts:1]: import type

File: controller/deliveryController.ts [hunk 4]
```diff
@@ -3,7 +3,19 @@ import { deliveryEmployee } from "../model/deliveryEmployee"
 
 import {createDeliveryEmployee, getDeliveryEmployeeByID, getDeliveryEmployees} from "../service/deliveryService"
 
+const validateAccess = function (req: Request, res: Response, next: Function) {
```
Comment (TomBillingtonUK) [inline controller/deliveryController.ts:6]: I would split this into the middleware folder

File: controller/deliveryController.ts [hunk 5]
```diff
@@ -3,7 +3,19 @@ import { deliveryEmployee } from "../model/deliveryEmployee"
...
+    if (req.session.current?.role === 'HR' || req.session.current?.role === 'SUPERUSER') {
+        next();
+    } else {
+        res.redirect('/forbidden');
+    }
+};
+
 export const deliveryController = (app:Application) =>{
+    app.use('/add-delivery-employee', validateAccess);
```
Comment (TomBillingtonUK) [inline controller/deliveryController.ts:15]: You could do this on the get/post. For example:

` app.get('/add-delivery-employee', validateAccess, async (req:Request, res: Response) => {`

If you had the code in the function as a method, your routes start looking like 

` app.get('/add-delivery-employee', validateAccess, addDeliverEmployeeGet)`

**Review by TomBillingtonUK — APPROVED** [review]

**Review by mateenparkar — APPROVED** [review]

**Pull request merged**